### PR TITLE
HomeWork04

### PR DESCRIPTION
**Pull request by YeaaaaH**

### CODE REVIEW

File: HW/src/main/java/homework04/ArrayList.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        container = Arrays.copyOf(container, size + value);
+    }
+    public void checkIndex(int index) {
+        if (index > this.size || index < 0) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + this.size);
+        }
+    }
+    @Override
+    public void add(T value) {
+        lengthIncrease(1);
```
Comment (Siril-O) [inline HW/src/main/java/homework04/ArrayList.java]: it is not efficient each add enlarge array to size+ 1 and copy elements, I suggest double it size when you have no place left for element

File: HW/src/main/java/homework04/ArrayList.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        container[size] = value;
+        size++;
+    }
+    @Override
+    public void add(T value, int index) {
+        checkIndex(index);
+        lengthIncrease(1);
+        T[] temporary = Arrays.copyOf(container, size);
+        container[index] = value;
```
Comment (Siril-O) [inline HW/src/main/java/homework04/ArrayList.java]: you will override existing element, you should add instead

File: HW/src/main/java/homework04/ArrayList.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    @Override
+    public void set(T value, int index) {
+        checkIndex(index);
+        container[index] = value;
+
+    }
+    @Override
+    public T remove(int index) {
+        checkIndex(index);
+        T buffer = container[index - 1];
```
Comment (Siril-O) [inline HW/src/main/java/homework04/ArrayList.java]: index in array should be the same as in list. list.get(0)  and array[0] not -1

Comment (YeaaaaH) [reply HW/src/main/java/homework04/ArrayList.java]: done

File: HW/src/main/java/homework04/ArrayList.java (hunk 1 above)
Comment (YeaaaaH) [reply HW/src/main/java/homework04/ArrayList.java]: done

**Pull request merged**